### PR TITLE
CIAC 6170 - Azure SQL Management

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-6170)

## Description
Make the parameters subscription_id, resource_group, workspace optional at the time of configuration and provide a subscription_id/Resource Group argument for all the relevant commands so that the default subscription id and resource_group at the time of configuration can be overridden and the integration instance can be flexible.
Add 2 new commands.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+#### Integrations
+
+##### Azure SQL Management
+
+- Updated the integration and took it out of beta.
+
+###### Command: 'azure-sql-db-list'
+- Added 2 new arguments 'subscription_id', 'resource_group_name' that if given will override the parameters that was given in the pack configuration.
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:9]: Please use the release notes template.
Group all changes together.
For example:
-  Added 2 new arguments *subscription_id* and *resource_group_name* that if given will override the parameters that were given in the pack configuration for the following commands:
  - list the commands

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+
+#### Integrations
+
+##### Azure SQL Management
+
+- Updated the integration and took it out of beta.
+
+###### Command: 'azure-sql-db-list'
+- Added 2 new arguments 'subscription_id', 'resource_group_name' that if given will override the parameters that was given in the pack configuration. 
+- Added values to the command's context data.
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:10]:
```suggestion
```
No need

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+###### Command: 'azure-sql-db-threat-policy-create-update'
+- Added 2 new arguments 'subscription_id', 'resource_group_name' that if given will override the parameters that was given in the pack configuration.
+- The argument 'resource_group_name' can be a list.
+
+###### New command: 'azure-sql-subscriptions-list'
+- Gets all the resource groups for a subscription.
+
+###### New command: 'azure-sql-resource-group-list'
+- Gets all the resource groups for a subscription.
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:39]: - Added new commands:
  - command 1
  - command 2

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 4]
```diff
@@ -49,57 +48,60 @@ In order to connect to the Azure SQL Management using either Cortex XSOAR Azure
...
+#### Command example
 ```!azure-sql-auth-start```
-
 #### Human Readable Output
 
 >### Authorization instructions
 >1. To sign in, use a web browser to open the page [https://microsoft.com/devicelogin](https://microsoft.com/devicelogin)
-> and enter the code **CODECODE** to authenticate.
->2. Run the ***!azure-sql-auth-complete*** command in the War Room.
+>and enter the code **DELTEABB7** to authenticate.
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:93]:
```suggestion
>and enter the code **CODECODE** to authenticate.
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 5]
```diff
@@ -355,27 +338,28 @@ Lists all of the databases for the server.
 >### Database List
 >|Name|Location|Status|Managed By|
 >|---|---|---|---|
->| master | eastus | Online | /subscriptions/0123456789/resourceGroups/sql-integration/providers/Microsoft.Sql/servers/sqlintegration |
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:358]: Revert this.

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py [hunk 6]
```diff
@@ -134,37 +141,70 @@ def azure_sql_db_threat_policy_create_update(self, server_name: str, db_name: st
...
                                                           f'/Microsoft.Sql/servers/{server_name}/databases/'
                                                           f'{db_name}/securityAlertPolicies/default',
                                  data=request_body)
 
+    def subscriptions_list_request(self):
+        """ Gets all subscriptions for a tenant.e.
+        Returns:
+            A dictionary that contains the list of subscription.
+        """
+        return self.http_request(method='GET', full_url='https://management.azure.com/subscriptions?api-version=2020-01-01')
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py:154]:
```suggestion
        return self.http_request(method='GET', full_url='https://management.azure.com/subscriptions?api-version=API_VERSION)
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py [hunk 7]
```diff
@@ -134,37 +141,70 @@ def azure_sql_db_threat_policy_create_update(self, server_name: str, db_name: st
...
+
+    def resource_group_list_request(self, sub_id: str, tag: str, limit: int):
+        """ Gets all the resource groups for a subscription.
+        Args:
+            sub_id: str - A subscription id.
+            tag: str - The tag and value that attached to the resource group.
+            limit: int - The number of results to return.
+        Returns:
+            A dictionary that contains the list of resource groups for the given subscription id.
+        """
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py:164]: Same here, use the constant for the API version

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py [hunk 8]
```diff
@@ -426,6 +479,81 @@ def azure_sql_db_threat_policy_create_update_command(client: Client, args: Dict[
...
+                'Provisioning State': result.get('properties', {}).get('provisioningState')
+            }
+            readable_output_table.append(d)
+        demisto.debug(f'{readable_output_table=}')
+        headers = ['Name', 'Location', 'Tags', 'Provisioning State']
+        human_readable = tableToMarkdown(name=f'Resource Group List for {sub_id}',
+                                         t=readable_output_table,
+                                         removeNull=True,
+                                         headers=headers)
+        demisto.debug(f'{human_readable=}')
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py:545]:
```suggestion
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py [hunk 9]
```diff
@@ -484,60 +664,66 @@ def main() -> None:
...
+
         if command == 'test-module':
+            client = create_client(params, subscription_id, resource_group_name)
             return_results(test_module(client))
 
-        elif command == 'azure-sql-servers-list':
-            return_results(azure_sql_servers_list_command(client, args))
+        elif command in commands_with_multiple_resource_group_name_list:
+            subscription_id = args.get('subscription_id') or params.get('subscription_id', '')
+            resource_group_name = argToList(args.get('resource_group_name')) or [params.get('resource_group_name', '')]
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py:684]: Why do we need this again?
You're getting it at the beginning of the try

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py [hunk 10]
```diff
@@ -484,60 +664,66 @@ def main() -> None:
...
-        )
+        subscription_id = args.get('subscription_id') or params.get('subscription_id', '')
+        resource_group_name = args.get('resource_group_name') or params.get('resource_group_name', '')
+        commands_with_multiple_resource_group_name_list = ['azure-sql-servers-list',
+                                                           'azure-sql-db-audit-policy-list',
+                                                           'azure-sql-db-audit-policy-create-update',
+                                                           'azure-sql-db-threat-policy-create-update']
+
         if command == 'test-module':
+            client = create_client(params, subscription_id, resource_group_name)
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py:679]: Why not call this once initially and then pass the client to the functions? 
Instead of creating the client for each command separately

**Review by Shellyber — COMMENTED** [review]
Nice Work

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 11]
```diff
@@ -116,36 +118,36 @@ There is no context output for this command.
...
 There is no context output for this command.
-
-#### Command Example
+#### Command example
 ```!azure-sql-auth-reset```
-
 #### Human Readable Output
 
->Authorization was reset successfully. You can now run ***!azure-sql-auth-start*** and ***!azure-sql-auth-complete***.
+>Authorization was reset successfully. You can now run **!azure-sql-auth-start** and **!azure-sql-auth-complete**.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:140]:
```suggestion
>Authorization was reset successfully. You can now run ***!azure-sql-auth-start*** and ***!azure-sql-auth-complete***.
```
Commands are bold and italics.

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 12]
```diff
@@ -162,20 +164,23 @@ There is no context output for this command.
...
+
 #### Input
 
 | **Argument Name** | **Description** | **Required** |
 | --- | --- | --- |
-| limit | The maximum number of servers returned to the War Room. Default is 50. | Optional | 
+| limit | (Int) The maximum number of servers returned to the War Room. Default is 50. | Optional | 
 | offset | Offset in the data set. Default is 0. | Optional | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:181]:
```suggestion
| subscription_id | Subscription ID. This value will override the instance param 'subscription_id'. | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 13]
```diff
@@ -162,20 +164,23 @@ There is no context output for this command.
...
 #### Input
 
 | **Argument Name** | **Description** | **Required** |
 | --- | --- | --- |
-| limit | The maximum number of servers returned to the War Room. Default is 50. | Optional | 
+| limit | (Int) The maximum number of servers returned to the War Room. Default is 50. | Optional | 
 | offset | Offset in the data set. Default is 0. | Optional | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional | 
+| resource_group_name | Resource group name. This value will override instance param 'resource_group_name'. Can support multiple resource groups, comma separated. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:182]:
```suggestion
| resource_group_name | A comma-separated list of resource group names. This value will override the instance param 'resource_group_name'.  | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 14]
```diff
@@ -230,21 +235,23 @@ Lists all the servers.
...
+
 #### Input
 
 | **Argument Name** | **Description** | **Required** |
 | --- | --- | --- |
 | server_name | Server name. | Required | 
 | limit | The maximum number of databases returned to the War Room. Default is 50. | Optional | 
 | offset | Offset in the data set. Default is 0. | Optional | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:253]:
```suggestion
| subscription_id | Subscription ID. This value will override the instance param 'subscription_id'. | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 15]
```diff
@@ -267,80 +274,56 @@ Lists all of the databases for the server.
...
-| AzureSQL.DB.readScale | String | The read-only routing state.  "Enabled" or "Disabled". | 
-| AzureSQL.DB.requestedServiceObjectiveName | String | The requested service objective name of the database.  | 
+| AzureSQL.DB.readScale | String | The read-only routing state. "Enabled" or "Disabled". | 
+| AzureSQL.DB.requestedServiceObjectiveName | String | The requested service objective name of the database. | 
 | AzureSQL.DB.status | String | Database status. | 
 | AzureSQL.DB.storageAccountType | String | Database storage account type. | 
 | AzureSQL.DB.zoneRedundant | Boolean | Whether the database zone is redundant. | 
+| AzureSQL.DB.currentBackupStorageRedundancy | Unknown | The storage account type used to store backups for this database. | 
+| AzureSQL.DB.requestedBackupStorageRedundancy | Unknown | The storage account type to be used to store backups for this database. | 
+| AzureSQL.DB.maintenanceConfigurationId | String | Maintenance configuration id assigned to the database. This configuration defines the period when the maintenance updates will occur. |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:284]:
```suggestion
| AzureSQL.DB.maintenanceConfigurationId | String | Maintenance configuration ID assigned to the database. This configuration defines the period when the maintenance updates will occur. | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 16]
```diff
@@ -267,80 +274,56 @@ Lists all of the databases for the server.
...
+| AzureSQL.DB.requestedServiceObjectiveName | String | The requested service objective name of the database. | 
 | AzureSQL.DB.status | String | Database status. | 
 | AzureSQL.DB.storageAccountType | String | Database storage account type. | 
 | AzureSQL.DB.zoneRedundant | Boolean | Whether the database zone is redundant. | 
+| AzureSQL.DB.currentBackupStorageRedundancy | Unknown | The storage account type used to store backups for this database. | 
+| AzureSQL.DB.requestedBackupStorageRedundancy | Unknown | The storage account type to be used to store backups for this database. | 
+| AzureSQL.DB.maintenanceConfigurationId | String | Maintenance configuration id assigned to the database. This configuration defines the period when the maintenance updates will occur. | 
+| AzureSQL.DB.isLedgerOn | Boolean | Whether or not this database is a ledger database, which means all tables in the database are ledger tables. The value of this property cannot be changed after the database has been created. | 
+| AzureSQL.DB.isInfraEncryptionEnabled | Boolean | Infra encryption is enabled for this database. |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:286]:
```suggestion
| AzureSQL.DB.isInfraEncryptionEnabled | Boolean | Whether infra encryption is enabled for this database. | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 17]
```diff
@@ -355,27 +338,28 @@ Lists all of the databases for the server.
...
 
 | **Argument Name** | **Description** | **Required** |
 | --- | --- | --- |
 | server_name | Server name. | Required | 
 | db_name | Database name. | Required | 
-| limit | The maximum number of DataBases audit policies returned to the War Room. Default is 50. | Optional | 
+| limit | The maximum number of database audit policies returned to the War Room. Default is 50. | Optional | 
 | offset | Offset in the data set. Default is 0. | Optional | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:361]:
```suggestion
| subscription_id | Subscription ID. This value will override the instance param 'subscription_id'. | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 18]
```diff
@@ -355,27 +338,28 @@ Lists all of the databases for the server.
...
 | **Argument Name** | **Description** | **Required** |
 | --- | --- | --- |
 | server_name | Server name. | Required | 
 | db_name | Database name. | Required | 
-| limit | The maximum number of DataBases audit policies returned to the War Room. Default is 50. | Optional | 
+| limit | The maximum number of database audit policies returned to the War Room. Default is 50. | Optional | 
 | offset | Offset in the data set. Default is 0. | Optional | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional | 
+| resource_group_name | Resource group name. This value will override instance param 'resource_group_name'. Can support multiple resource groups, comma separated. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:362]:
```suggestion
| resource_group_name | A comma-separated list of resource group names. This value will override the instance param 'resource_group_name'.  | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 19]
```diff
@@ -386,17 +370,17 @@ Gets the audit settings of the specified database.
...
 | AzureSQL.DBAuditPolicy.name | String | Audit policy name. | 
 | AzureSQL.DBAuditPolicy.type | String | Resource type. | 
-| AzureSQL.DBAuditPolicy.isAzureMonitorTargetEnabled | Boolean | Whether audit events are sent to Azure Monitor. Possible values: "True" (Enabled) or "False" (Disabled). | 
+| AzureSQL.DBAuditPolicy.isAzureMonitorTargetEnabled | Boolean | Whether audit events are sent to Azure Monitor. Possible values: "True" \(Enabled\) or "False" \(Disabled\). | 
 | AzureSQL.DBAuditPolicy.retentionDays | Number | Number of days to keep in the audit logs in the storage account. | 
 | AzureSQL.DBAuditPolicy.state | String | Policy state. | 
 | AzureSQL.DBAuditPolicy.storageAccountSubscriptionId | String | Storage subscription ID. | 
 | AzureSQL.DBAuditPolicy.databaseName | String | The name of the database that the audit policy is related to. | 
 | AzureSQL.DBAuditPolicy.serverName | String | The name of the server that the audit policy is related to. | 
+| AzureSQL.DBAuditPolicy.isManagedIdentityInUse | Boolean | Specifies whether Managed Identity is used to access blob storage |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:379]:
```suggestion
| AzureSQL.DBAuditPolicy.isManagedIdentityInUse | Boolean | Specifies whether Managed Identity is used to access blob storage. | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 20]
```diff
@@ -425,27 +409,29 @@ Gets the audit settings of the specified database.
...
 `azure-sql-db-threat-policy-get`
+
 #### Input
 
 | **Argument Name** | **Description** | **Required** |
 | --- | --- | --- |
 | server_name | Server name. | Required | 
 | db_name | Database name. | Required | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:433]:
```suggestion
| subscription_id | Subscription ID. This value will override the instance param 'subscription_id'. | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 21]
```diff
@@ -425,27 +409,29 @@ Gets the audit settings of the specified database.
...
+
 #### Input
 
 | **Argument Name** | **Description** | **Required** |
 | --- | --- | --- |
 | server_name | Server name. | Required | 
 | db_name | Database name. | Required | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional | 
+| resource_group_name | Resource group name. This value will override instance param 'resource_group_name'. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:434]:
```suggestion
| resource_group_name | Resource group name. This value will override the instance param 'resource_group_name'. | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 22]
```diff
@@ -531,8 +515,10 @@ Creates or updates the database's auditing policy.
 | retention_days | Number of days to keep the policy in the audit logs. | Optional | 
 | storage_account_access_key | Identifier key of the auditing storage account. | Optional | 
 | storage_account_subscription_id | Storage subscription ID. | Optional | 
-| storage_endpoint | Storage endpoint. If the value for the state argument is enabled, the value for the &storage_endpoint* or *is_azure_monitor_target_enabled* argument is required. | Optional | 
-
+| storage_endpoint | Storage endpoint. If the value for the state argument is enabled, the value for the storage_endpoint or is_azure_monitor_target_enabled argument is required. | Optional | 
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:519]:
```suggestion
| subscription_id | Subscription ID. This value will override the instance param 'subscription_id'. | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 23]
```diff
@@ -531,8 +515,10 @@ Creates or updates the database's auditing policy.
 | retention_days | Number of days to keep the policy in the audit logs. | Optional | 
 | storage_account_access_key | Identifier key of the auditing storage account. | Optional | 
 | storage_account_subscription_id | Storage subscription ID. | Optional | 
-| storage_endpoint | Storage endpoint. If the value for the state argument is enabled, the value for the &storage_endpoint* or *is_azure_monitor_target_enabled* argument is required. | Optional | 
-
+| storage_endpoint | Storage endpoint. If the value for the state argument is enabled, the value for the storage_endpoint or is_azure_monitor_target_enabled argument is required. | Optional | 
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional | 
+| resource_group_name | Resource group name. This value will override instance param 'resource_group_name'. Can support multiple resource groups, comma separated. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:520]:
```suggestion
| resource_group_name | A comma-separated list of resource group names. This value will override the instance param 'resource_group_name'.  | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 24]
```diff
@@ -607,39 +594,38 @@ Creates or updates the database's threat detection policy.
...
 | retention_days | Number of days to keep the policy in the audit logs. | Optional | 
-| storage_account_access_key | The identifier key of the threat detection audit storage account | Optional | 
+| storage_account_access_key | The identifier key of the threat detection audit storage account. | Optional | 
 | storage_endpoint | The blob storage endpoint. This blob storage will hold all Threat Detection audit logs. | Optional | 
 | disabled_alerts | Comma-separated list of alerts that are disabled. Possible values: "None", "Sql_Injection", "Sql_Injection_Vulnerability", "Access_Anomaly", "Data_Exfiltration", and "Unsafe_Action". | Optional | 
 | email_addresses | Comma-separated list of email addresses to which the alert is sent. | Optional | 
 | email_account_admins | Whether the alert is sent to the account administrators. Possible values: "true" and "false". | Optional | 
 | use_server_default | Whether to use the default server policy. Possible values: "Enabled" and "Disabled". | Optional | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:603]:
```suggestion
| subscription_id | Subscription ID. This value will override the instance param 'subscription_id'. | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 25]
```diff
@@ -607,39 +594,38 @@ Creates or updates the database's threat detection policy.
...
+| storage_account_access_key | The identifier key of the threat detection audit storage account. | Optional | 
 | storage_endpoint | The blob storage endpoint. This blob storage will hold all Threat Detection audit logs. | Optional | 
 | disabled_alerts | Comma-separated list of alerts that are disabled. Possible values: "None", "Sql_Injection", "Sql_Injection_Vulnerability", "Access_Anomaly", "Data_Exfiltration", and "Unsafe_Action". | Optional | 
 | email_addresses | Comma-separated list of email addresses to which the alert is sent. | Optional | 
 | email_account_admins | Whether the alert is sent to the account administrators. Possible values: "true" and "false". | Optional | 
 | use_server_default | Whether to use the default server policy. Possible values: "Enabled" and "Disabled". | Optional | 
-
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. | Optional | 
+| resource_group_name | Resource group name. This value will override instance param 'resource_group_name'. Can support multiple resource groups, comma separated. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:604]:
```suggestion
| resource_group_name | A comma-separated list of resource group names. This value will override the instance param 'resource_group_name'.  | Optional | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 26]
```diff
@@ -607,39 +594,38 @@ Creates or updates the database's threat detection policy.
...
+| resource_group_name | Resource group name. This value will override instance param 'resource_group_name'. Can support multiple resource groups, comma separated. | Optional | 
 
 #### Context Output
 
 | **Path** | **Type** | **Description** |
 | --- | --- | --- |
 | AzureSQL.DBThreatPolicy.kind | String | Kind of threat policy. | 
 | AzureSQL.DBThreatPolicy.location | String | Threat policy location. | 
-| AzureSQL.DBThreatPolicy.id | String | Threat policy ID. | 
+| AzureSQL.DBThreatPolicy.id | String | Threat policy ID |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:612]:
```suggestion
| AzureSQL.DBThreatPolicy.id | String | Threat policy ID. | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 27]
```diff
@@ -607,39 +594,38 @@ Creates or updates the database's threat detection policy.
...
 | AzureSQL.DBThreatPolicy.storageAccountAccessKey | String | The identifier key of the Threat Detection audit storage account. | 
 | AzureSQL.DBThreatPolicy.storageEndpoint | String | Threat Detection audit storage account. | 
-| AzureSQL.DBThreatPolicy.emailAccountAdmins | Boolean | Email account administrators who the alert is sent to. | 
+| AzureSQL.DBThreatPolicy.emailAccountAdmins | Boolean | Email accounts administrators who the alert is sent to. | 
 | AzureSQL.DBThreatPolicy.emailAddresses | String | List of email addresses to which the alert is sent. | 
 | AzureSQL.DBThreatPolicy.disabledAlerts | String | List of alerts that are disabled, or an empty string if no alerts are disabled. | 
 | AzureSQL.DBThreatPolicy.useServerDefault | Unknown | Whether to use the default server policy. | 
 | AzureSQL.DBThreatPolicy.databaseName | String | The name of the database that the threat policy is related to. | 
-| AzureSQL.DBThreatPolicy.serverName | String | The name of the server that the threat policy is related to. | 
+| AzureSQL.DBThreatPolicy.serverName | String | The name of server that the threat policy is related to. |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:625]:
```suggestion
| AzureSQL.DBThreatPolicy.serverName | String | The name of the server that the threat policy is related to. | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 28]
```diff
@@ -701,4 +688,118 @@ You will be automatically redirected to a link with the following structure:
...
+| **Path** | **Type** | **Description** |
+| --- | --- | --- |
+| AzureSQL.Subscription.id | String | The fully qualified ID for the subscription. | 
+| AzureSQL.Subscription.authorizationSource | String | The authorization source of the request. Valid values are one or more combinations of Legacy, RoleBased, Bypassed, Direct and Management. For example, 'Legacy, RoleBased'. | 
+| AzureSQL.Subscription.managedByTenants | Unknown | An array containing the tenants managing the subscription. | 
+| AzureSQL.Subscription.subscriptionId | String | The subscription ID. | 
+| AzureSQL.Subscription.tenantId | String | The subscription tenant ID. | 
+| AzureSQL.Subscription.displayName | String | The subscription display name. | 
+| AzureSQL.Subscription.state | String | The subscription state. Possible values are Enabled, Warned, PastDue, Disabled, and Deleted. | 
+| AzureSQL.Subscription.subscriptionPolicies.locationPlacementId | String | The subscription location placement ID. The ID indicates which regions are visible for a subscription. For example, a subscription with a location placement Id of Public_2014-09-01 has access to Azure public regions. |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:715]:
```suggestion
| AzureSQL.Subscription.subscriptionPolicies.locationPlacementId | String | The subscription location placement ID. The ID indicates which regions are visible for a subscription. For example, a subscription with a location placement ID of Public_2014-09-01 has access to Azure public regions. | 
```

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 29]
```diff
@@ -701,4 +688,118 @@ You will be automatically redirected to a link with the following structure:
...
+
+#### Base Command
+
+`azure-sql-resource-group-list`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| subscription_id | Subscription id. This value will override instance param 'subscription_id'. Can support multiple subscription ids, comma separated. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:763]:
```suggestion
| subscription_id | A comma-separated list of subscription IDs. This value will override the instance param 'subscription_id'. | Optional | 
```

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 30]
```diff
@@ -0,0 +1,39 @@
+
+#### Integrations
+
+##### Azure SQL Management
+
+- Updated the integration and took it out of beta.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:6]:
```suggestion
Updated the integration and took it out of beta.
```

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 31]
```diff
@@ -0,0 +1,39 @@
...
+
+###### Command: 'azure-sql-db-list'
+- Added 2 new arguments 'subscription_id', 'resource_group_name' that if given will override the parameters that was given in the pack configuration. 
+- Added values to the command's context data.
+
+###### Command: 'azure-sql-servers-list'
+- Added 2 new arguments 'subscription_id', 'resource_group_name' that if given will override the parameters that was given in the pack configuration.
+- The argument 'resource_group_name' can be a list.
+- Added an argument 'list_by_resource_group' that if set to 'true' will return the list of servers that matches to the given resource_group_name. 
+- Values were to the command's context data.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:16]:
```suggestion
- Values were added to the command's context data.
```
Perhaps you should list the values here?

**Comment by ShirleyDenkberg** [comment]
@RotemAmit I agree with Shelly's comments in the RN.  If you want, I can review the RN when you are done implementing her changes. Just ping me when you are done.
@Shellyber My doc review was completed.

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 32]
```diff
@@ -0,0 +1,17 @@
+
+#### Integrations
+
+##### Azure SQL Management
+
+- Updated the integration and took it out of beta.
+- Added 2 new arguments subscription_id and resource_group_name that if given will override the parameters that were given in the pack configuration for the following commands:
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:8]:
```suggestion
- Added the subscription_id and resource_group_name arguments to the following commands. If these arguments are provided, they will override the parameters that were provided in the pack configuration.
```

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 33]
```diff
@@ -0,0 +1,17 @@
...
+  - azure-sql-db-list
+  - azure-sql-servers-list
+  - azure-sql-db-audit-policy-list
+  - azure-sql-db-audit-policy-create-update
+  - azure-sql-db-threat-policy-get
+  - azure-sql-db-threat-policy-create-update
+- Updated the configuration parameters 'Subscription ID' and 'Resource Group Name' to be not required, and updated their names to  'Default Subscription ID' and 'Default Resource Group Name'.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:15]:
```suggestion
- Updated the configuration parameters 'Subscription ID' and 'Resource Group Name' to be not required, and updated their names to 'Default Subscription ID' and 'Default Resource Group Name'.
```

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 34]
```diff
@@ -0,0 +1,17 @@
...
+- Added new commands:
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:16]:
```suggestion
- Added the following new commands:
```

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md [hunk 35]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_18.md:1]: Revert this change

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py [hunk 36]
```diff
@@ -538,9 +688,19 @@ def main() -> None:
...
+        elif command == 'azure-sql-subscriptions-list':
+            return_results(subscriptions_list_command(client))
+
+        elif command == 'azure-sql-resource-group-list':
+            return_results(resource_group_list_command(client, args, subscription_id))
+
     # Log exceptions and return errors
     except Exception as e:
-        return_error(f'Failed to execute {demisto.command()} command.\nError:\n{str(e)}')
+        if 'Error in API call [404] - Not Found' in e.message:
```
Comment (Shellyber) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement.py:699]: I rather this to be in the def http_request() function.
Check what returns from the response and then do this check.
If status_code == 404 and 'Not Found':
return message.

**Review by Shellyber — APPROVED** [review]

**Comment by RotemAmit** [comment]
All the steps in code.pan.run passed except the xsoar_ng step. The error in this step is: "no space left on device", "could not reterive docker image".
